### PR TITLE
Add option to make screen markers not click through

### PR DESCRIPTION
**Pull request by MagicfTail**
Add an option to make the screen markers not click through, similar to osbuddys screen markers.

Note, the icons are very much placeholders

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ScreenMarkerPlugin.java [hunk 1]
```diff
@@ -124,6 +124,8 @@ protected void startUp() throws Exception
 		clientToolbar.addNavigation(navigationButton);
 
 		mouseListener = new ScreenMarkerMouseListener(this);
+
+		mouseManager.registerMouseListener(mouseListener);
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ScreenMarkerPlugin.java]: You would need to unregister this mouse listener in shutdown.

File: runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ScreenMarkerPlugin.java [hunk 2]
```diff
@@ -229,6 +225,18 @@ void resizeMarker(Point point)
 		overlay.setPreferredSize(bounds.getSize());
 	}
 
+	public boolean shouldEatClick(Point point)
+	{
+		for (ScreenMarkerOverlay screenMarker : screenMarkers)
+		{
+			if (screenMarker.getBounds().contains(point) && !screenMarker.getMarker().isClickThrough())
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ScreenMarkerPlugin.java]: check if the marker is not click trough first

**Comment by Kamielvf** [comment]
Have you tested how this behaves when loading previously stored screen markers?

**Comment by MagicfTail** [comment]
It sets all existing screen markers to no click through, which is a bit unfortunate

**Comment by Kamielvf** [comment]
In that case I would make it so that it is click trough by default to not change the existing behavior.

**Comment by MagicfTail** [comment]
Existing screen markers are now click through

**Comment by Kamielvf** [comment]
I think it would be better to locate this button next to the visible toggle

File: runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ScreenMarkerPlugin.java (hunk 1 above)
Comment (MagicfTail) [reply runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ScreenMarkerPlugin.java]: Done

**Comment by MagicfTail** [comment]
It's now moved to the right and rebased

**Pull request closed without merging**